### PR TITLE
Romcom fp - M1 paired project

### LINKED ISSUE #582: Update paired project 2
# Project to be updated: [Rom Com](https://frontend.turing.edu/projects/module-1/romcom-pair-v2.html)

## Summary of changes needed: 
- The project currently pushes students to make use of a Cover class, this should be replaced with some guidance on how to store the information about a cover in an object literal
- Reference to using the Cover class should likewise be updated
- The boilerplate Cover class should be removed from the starter repo
- All group projects should make clear that a failure to participate can result in an individual failing the project, even if the project is otherwise passing

### PR DESCRIPTION
**Pull request by hfaerber**
Changes Made:
- Removed classes
- Added note that individual may fail otherwise passing project if not contributing/collaborating
- Added purity to learning goals and rubric

- Made a few minor edits to already updated Hang In There project to keep the two consistent with each other

Questions for reviewer:
- Please also review [this PR](https://github.com/turingschool-examples/romcom/pull/324) for updates to the starter-repo.  I have questions about how we want/expect students to organize their code between the two files.
- I put purity as a WOW under the Functional Programming section instead of in the JS section. Does WOW feel appropriate for purity in M1?  Does FP feel like the right section?

Will close #582

### CODE REVIEW

File: projects/module-1/romcom-pair-v2.md [hunk 1]
```diff
@@ -128,6 +124,7 @@ Here's a list of possible extensions to implement - but **ONLY IF** your team ha
 You are welcome to add your own extensions. Be sure they are thoughtful in terms of UX/UI, and that they do not break any prior functionality.
 
 - Implement data validation and error handling into the form (disable the button until all fields are filled, provide error messages if data entered is not correct, etc)
+- In the main cover view, allow users to click each piece of the cover (image, title, descriptors) to update just that piece with another random item from the appropriate array
```
Comment (CassandraGoose) [inline projects/module-1/romcom-pair-v2.md:126]: Love this extension addition!

File: projects/module-1/romcom-pair-v2.md [hunk 2]
```diff
@@ -203,16 +200,16 @@ This project gives the team room to practice the following JavaScript concepts:
...
+### Functional Programing competency examples
 
-- Project makes use of the provided class, creating instances to encapsulate & store unique data
-- Team considers when and how to create new instances or update existing instances, coming to a clear and consistent decision
-- Team effectively creates/accesses instance properties when applicable
-- Team effectively invokes instance methods when applicable
+- Project makes use of the provided `createCover` function, creating objects to encapsulate & store unique data
+- Team considers when and how to create new cover objects or update existing cover objects, coming to a clear and consistent decision
+- Team effectively creates/accesses object properties when applicable
+- Team effectively invokes object methods when applicable
```
Comment (CassandraGoose) [inline projects/module-1/romcom-pair-v2.md]: I wonder if this is going to encourage them to create object methods? I think it's good to know how to do it in general, but I'm not sure if it counts as a functional programming competency. Thoughts?

File: projects/module-1/romcom-pair-v2.md [hunk 3]
```diff
@@ -270,9 +267,12 @@ Competency & understanding of these concepts can be demonstrated in many ways. *
 
 - The team's norming document is regularly revisted and revised as better perspective is gained
 - When conflicts arise, the team works together to find ways to re-focus on individual and group learning objectives
-</section>
 
 Note: ⬆️this⬆️ kind of collaboration can be difficult to assess, but know that this is a crucial and vital part of your learning and marketability as a developer!
```
Comment (CassandraGoose) [inline projects/module-1/romcom-pair-v2.md:268]: Love the addition of this note!

**Review by CassandraGoose — COMMENTED** [review]
I think that there's a leftover note that relates to classes in the "Reading Code competency examples" section. Not sure what line of code it is, but it reads: "Provided code is used effectively (the provided Class, the provided assets, the provided HTML & CSS)"

File: projects/module-1/romcom-pair-v2.md (hunk 2 above)
Comment (hfaerber) [reply projects/module-1/romcom-pair-v2.md]: Thats a really great callout.  Honestly these bullets are pretty directly tranferred from when this section was OOP.  I might need some help brainstorming more appropriate bullets for this section

**Comment by hfaerber** [comment]
> 

Ah thank you for catching this!  I've removed the class reference and referenced the createCover function instead (in both romcom and hang in there)

**Comment by sertmer** [comment]
I think this looks great Heather! I think the function purity is a perfect fit for WOW!

**Pull request merged**